src/operator/src/expr_factory.rs

```diff
@@ -189,7 +189,8 @@ pub fn create_to_expr(create: &CreateTable, query_ctx: QueryContextRef) -> Resul
 
     let time_index = find_time_index(&create.constraints)?;
     let table_options = HashMap::from(
-        &TableOptions::try_from(create.options.as_ref()).context(UnrecognizedTableOptionSnafu)?,
+        &TableOptions::try_from(&create.options.clone().into_map())
```

Comment:
Extra clone. This is subtle because it's awkward to convert between `HashMap<&String, &String>` and `&HashMap<String, String>`.

Maybe we can duplicate the `try_from` impl for `HashMap<&String, &String>`, but I don't like this idea actually.